src/docs/arch-transaction.dox

```diff
@@ -5,4 +5,114 @@ A caller of WiredTiger uses @ref transactions within the API to start and stop t
 a session (thread of control).
 
 Internally, the current transaction state is represented by the WT_TXN structure.
+
+WiredTiger performs all the operations in transactions. If the user doesn't explicitly declare a transaction, WiredTiger will automatically create a transaction for the user's operation.
```

Comment:
Instead of declare I would say something like "if the user doesn't explicitly begin a transaction"

Reply:
Another comment on this line, I think we should be specific about which operations WiredTiger will internally begin a transaction for, do we start one for a single cursor read? If so which isolation level will it run at. These things I think will be helpful to have specified.

Reply:
Some of them are mentioned later in the isolation section.